mlflow/tracking/client.py

```diff
@@ -2116,7 +2116,12 @@ def create_model_version(
             Stage: None
         """
         tracking_uri = self._tracking_client.tracking_uri
-        if not run_link and is_databricks_uri(tracking_uri) and tracking_uri != self._registry_uri:
+        if (
+            not run_link
+            and run_id
+            and is_databricks_uri(tracking_uri)
+            and tracking_uri != self._registry_uri
+        ):
```

Comment:
How about:
```suggestion
        if (
            not run_link
            and is_databricks_uri(tracking_uri)
            and tracking_uri != self._registry_uri
        ):
          if not run_id:
            eprint("Warning: no run_link will be recorded with the model version because no run_id was given") 
          else:
            run_link = ...
```